README.md

```diff
@@ -29,20 +29,43 @@ The Dapr docs are built using [Hugo](https://gohugo.io/) with the [Docsy](https:
 
 The [daprdocs](./daprdocs) directory contains the hugo project, markdown files, and theme configurations.
 
-## Pre-requisites
+## Setup with a devcontainer
 
-- [Hugo extended version](https://gohugo.io/getting-started/installing)
-- [Node.js](https://nodejs.org/en/)
+This repository comes with a [devcontainer](/.devcontainer/devcontainer.json) configuration that automatically installs all the required dependencies and tools to build and run the docs.
+
+This devcontainer can be used to develop locally with VSCode or via GitHub Codespaces completely in the browser.
+
+### Pre-requisites
 
-## Environment setup
+- [Docker Desktop](https://www.docker.com/products/docker-desktop)
+- [VSCode](https://code.visualstudio.com/download)
 
-1. Ensure pre-requisites are installed
-2. Clone this repository
+### Environment setup
+
+1. [Fork](https://github.com/dapr/docs/fork) and clone this repository.
+
+2. Open the forked repository in VS Code
 
 ```sh
-git clone https://github.com/dapr/docs.git
+code .
 ```
 
+3. When prompted, click "Reopen in Container" to open the repository in the devcontainer.
```

Comment:
I tried your PR out with Goland and didn't get prompted. For VS Code I did get the prompt. It might be worth documenting the steps a user should take if they don't get prompted by their IDE. 

It also took quite a while to load and get to this point, specifically, it took: 3.3588 minutes for it to start my devcontainer for me to be able to proceed:
```
[201528 ms] Start: Run in container: cd '/vscode/vscode-server/extensionsCache' && ls -t | tail -n +50 | xargs rm -f
[201532 ms] 
[201532 ms] 
[238758 ms] Port forwarding connection from 49714 > 41037 > 41037 in the container.
[238758 ms] Start: Run in container: /home/vscode/.vscode-server/bin/0ee08df0cf4527e40edc9aa28f4b5bd38bbff2b2/node -e 
[238893 ms] Port forwarding 49714 > 41037 > 41037 stderr: Connection established
[243903 ms] Port forwarding 49714 > 41037 > 41037 stderr: Remote close
[243917 ms] Port forwarding 49714 > 41037 > 41037 terminated with code 0 and signal null.
[376402 ms] Port forwarding 49714 > 41037 > 41037: Local close
```

Reply:
Was this container running locally or on GitHub?

Reply:
I followed the instructions provided in the readme- I ran this locally. I wan't sure if @marcduiker's also took a while? It's just something I wanted to note since it does take a bit more time now.

Reply:
@cicoyle When I try it with VSCode it takes about 26 seconds (I've deleted the container and the image first). When I close and reopen VSCode with the container it takes about 8 seconds.

The devcontainer support for JetBrains products is still very new (was announced Dec 2023) so perhaps they need to improve their tooling a bit?